src/libraries/Common/src/Interop/Windows/Kernel32/Interop.DeviceIoControl.cs

```diff
@@ -15,6 +15,45 @@ internal static partial class Kernel32
         // https://docs.microsoft.com/windows-hardware/drivers/ddi/ntddstor/ni-ntddstor-ioctl_storage_read_capacity
         internal const int IOCTL_STORAGE_READ_CAPACITY = 0x002D5140;
 
+        // https://learn.microsoft.com/windows/win32/api/winioctl/ni-winioctl-fsctl_set_sparse
+        internal const int FSCTL_SET_SPARSE = 0x000900c4;
+        internal readonly struct FILE_SET_SPARSE_BUFFER
```

Comment:
The existing control codes in this file have the supporting structures in separate files. It would be nice to follow consistent style.

Reply:
The other control codes in this files don't have structures. Did you still want me to extract them? What would I call it?

Reply:
> The other control codes in this files don't have structures (I think). 

They do:
`FSCTL_GET_REPARSE_POINT` returns [several different structures](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.REPARSE_DATA_BUFFER.cs)
`IOCTL_STORAGE_READ_CAPACITY` return (`STORAGE_READ_CAPACITY`)(https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.STORAGE_READ_CAPACITY.cs)

Reply:
Did I need to specify sequential layout, like `STORAGE_READ_CAPACITY` does?

Reply:
`struct` definitions in C# have sequential layout by default, so it is not necessary to specify it for correctness.

Specifying sequential layout has a side-effect of suppressing "unused field" warnings from C# compiler. People like to explicitly specify sequential layout to get rid of these warnings. It looks better than suppressing the warnings using `#pragma`.

Reply:
Interesting, I didn't know it did that. Good to know